hera_cal/apply_cal.py

```diff
@@ -21,6 +23,17 @@ def recalibrate_in_place(data, data_flags, new_gains, cal_flags, old_gains=None,
         gain_convention: str, either 'divide' or 'multiply'. 'divide' means V_obs = gi gj* V_true,
             'multiply' means V_true = gi gj* V_obs. Assumed to be the same for new_gains and old_gains.
     '''
+    # get datatype of data_flags to determine if flags or wgts
+    if data_flags[data_flags.keys()[0]].dtype == np.bool:
```

Comment:
I'd recommend something like `if np.all([(df.dtype == np.bool) for df in data.values()]):` since it checks all entries and doesn't really take that much longer.